frontend/src/components/RecordsList.tsx

```diff
@@ -120,6 +129,21 @@ const RecordsList: FunctionComponent<IRecordsListProps> = ({
     }
   };
 
+  const handleSearch = async () => {
+    const uniqueQueries = parseSearchQueries(val);
+    if (
+      customFilterUI &&
+      customFilterState &&
+      setCustomFilterVals!.length > 0 &&
+      customFilterFunc
+    ) {
+      const newQueries = await customFilterFunc(uniqueQueries);
+      setSearchVal(newQueries);
+    } else {
+      setSearchVal(uniqueQueries);
+    }
+  };
+
```

Comment:
How this works in a nutshell:
- If the user is performing a standard search/filter (with MRN mode off), then everything carries on as before
- However, if they're searching with MRNs, then we'll fetch from the protected /mrn-search API endpoint, receive and parse out the corresponding CMO Patient IDs, then use those CMO Patient IDs to search/filter as normal